### PR TITLE
feat: enable via_ir in solidity settings [APE-1256]

### PR DESCRIPTION
**Pull request by z80dev**
### What I did

Enable compiling contracts with the `--via-ir` solc flag

### How I did it

parse a boolean config value for enabling the `--via-ir` flag. this enables compiling solidity contracts through the Yul IR pipeline. This avoids `stack too deep` errors.

I wrote the affected contract as inline code because if I write it as a contract in the contracts folder, then we have to enable `via_ir` for the whole project, or else all tests fail when compilation is attempted

so I wrote it as inline source, write it to a file, attempt compilation without `via_ir` and check that it fails, and then re-attempt compilation after enabling the  `via_ir` flag.

### How to verify it

Add the following config values in a repo with contracts affected by stack-too-deep

```yaml
solidity:
    via_ir: True
```

### Checklist

- [x] Passes all linting checks (pre-commit and CI jobs)
- [x] New test cases have been added and are passing
- [x] Documentation has been updated
- [x] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: tests/test_compiler.py [hunk 1]
```diff
@@ -438,6 +439,72 @@ def test_ast(project, compiler):
...
+
+        return (sum);
+    }
+
+}
+    """
+
+    # write source code to file
+    with open(source_path, "w") as f:
+        f.write(source_code)
```
Comment (fubuloubu) [inline tests/test_compiler.py:490]: `pathlib.Path` has this method which handles context management of file handles for you
```suggestion
   source_path.write_text(source_code)
```

File: tests/test_compiler.py [hunk 2]
```diff
@@ -438,6 +439,72 @@ def test_ast(project, compiler):
...
+        compiler.compile([source_path])
+    except Exception as e:
+        assert "Stack too deep" in str(e)
+
+    compiler.config.via_ir = True
+
+    compiler.compile([source_path])
+
+    # delete source code file
+    os.remove(source_path)
```
Comment (fubuloubu) [inline tests/test_compiler.py:502]: `pathlib.Path` equivalent
```suggestion
    source_path.unlink()
```

https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module

**Review by fubuloubu — DISMISSED** [review]
Looks good otherwise

**Review by fubuloubu — APPROVED** [review]

File: tests/test_compiler.py [hunk 3]
```diff
@@ -438,6 +438,71 @@ def test_ast(project, compiler):
...
+    source_path.write_text(source_code)
+
+    try:
+        compiler.compile([source_path])
+    except Exception as e:
+        assert "Stack too deep" in str(e)
+
+    compiler.config.via_ir = True
+
+    compiler.compile([source_path])
```
Comment (antazoey) [inline tests/test_compiler.py:497]: it might be nice to assert something about the result here

**Pull request merged**